File: micronaut-java-parent-example/domain/src/main/java/io/archimedesfw/maven/micronaut/example/todo/Todo.java
package io.archimedesfw.maven.micronaut.example.todo;

import io.micronaut.core.annotation.Nullable;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.serde.annotation.Serdeable;

import java.time.LocalDateTime;

@MappedEntity
@Serdeable
public record Todo(
        @Id
        @GeneratedValue
        @Nullable Integer id,

        String title,
        LocalDateTime creationDate,
        @Nullable LocalDateTime doneDate
) {}

File: micronaut-kotlin-parent-example/domain/src/main/kotlin/io/archimedesfw/maven/micronaut/example/todo/Todo.kt
package io.archimedesfw.maven.micronaut.example.todo

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.serde.annotation.Serdeable
import java.time.LocalDateTime

@Serdeable
@MappedEntity
data class Todo(
    @field:Id
    @field:GeneratedValue
    val id: Int? = null,

    val title: String,
    val creationDate: LocalDateTime,
    val doneDate: LocalDateTime? = null
)
